modules/Bio/EnsEMBL/Compara/RunnableDB/GeneTrees/OrthoTree.pm

```diff
@@ -566,5 +602,115 @@ sub check_homology_consistency {
     $self->throw("Inconsistent homologies: $bad_key") if defined $bad_key;
 }
 
+sub _create_flatfile {
+    my $self = shift;
+
+    my $outfile = $self->param('output_flatfile');
+
+    # create directory
+    my $outdir = dirname($outfile);
+    $self->run_command("mkdir -p $outdir");
```

Comment:
I know it is done this way in several other runnables, but `run_command` will disconnect from the database. Assuming the rest of `run_analysis` needs to fetch some data, that's going to be a very short disconnection cycle. I would use
```perl
use File::Path qw(make_path)

make_path($outdir);
```